frontend/src/components/dashboard/MyOrganizationTabContent.tsx

```diff
@@ -10,47 +11,206 @@ import {
   PaginationPrevious,
 } from "@components/ui/pagination";
 import { SelectableItemsList } from "@components/ui/selectable-items-list";
-import { DocumentReportIcon, EyeOffIcon, GlobeAltIcon } from "@lib/icons";
+import { Dataset } from "@interfaces/ckan/dataset.interface";
+import {
+  DocumentReportIcon,
+  DocumentSearchIcon,
+  EyeOffIcon,
+  GlobeAltIcon,
+} from "@lib/icons";
+import { SearchPageOnChange } from "@pages/search";
 import { SearchDatasetType } from "@schema/dataset.schema";
 import { api } from "@utils/api";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
 export default () => {
-  const [page, setPage] = useState(1);
-  const [visibility, setVisibility] = useState("All");
-  const [contributor, setContributor] = useState("All");
+  const [contributors, setContributors] = useState<Facet[]>([]);
```

Comment:
Maybe move the states into a custom hook or component so that we can reuse that both on My Datasets and My Organizations so that if we have to do fixes or new features on the search pages it's easier to maintain